ckanext/dcat/profiles.py

```diff
@@ -769,6 +777,19 @@ def parse_dataset(self, dataset_dict, dataset_ref):
 
             dataset_dict['resources'].append(resource_dict)
 
+            # License
+            if 'license_id' not in dataset_dict:
+                # If distribution has a license, attach it to the dataset
+                license = self._object(distribution, DCT.license)
+                if license:
+                    # Try to find a matching license comparing URIs, then titles
+                    license_id = license_uri2id.get(license.toPython())
+                    if license_id is None:
+                        license_id = license_title2id.get(
+                            self._object_value(license, DCT.title))
+                    if license_id is not None:
+                        dataset_dict['license_id'] = license_id
+
```

Comment:
I would propose to put this whole block in a method called `_license()` to make it very easy to override this behavior.